hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -99,25 +99,30 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		if err != nil {
 			return ctrl.Result{}, fmt.Errorf("failed to obtain AMI: %w", err)
 		}
-		machineSet, _, err := generateMachineScalableResources(hcluster.Spec.InfraID, ami, nodePool, targetNamespace)
+		machineDeployment, _, err := generateMachineScalableResources(hcluster.Spec.InfraID, ami, nodePool, targetNamespace)
 		if err != nil {
 			return reconcile.Result{}, fmt.Errorf("failed to generate worker machineset: %w", err)
 		}
-		if err := r.Delete(ctx, machineSet); err != nil && !apierrors.IsNotFound(err) {
+		if err := r.Delete(ctx, machineDeployment); err != nil && !apierrors.IsNotFound(err) {
 			return reconcile.Result{}, fmt.Errorf("failed to delete nodePool: %w", err)
 		}
 		mcs := generateMachineConfigServer(nodePool, targetNamespace)
 		if err := r.Delete(ctx, mcs); err != nil && !apierrors.IsNotFound(err) {
 			return reconcile.Result{}, fmt.Errorf("failed to delete machineConfigServer: %w", err)
 		}
 
+		mhc := generateMachineHealthCheck(nodePool, targetNamespace, hcluster.Spec.InfraID)
+		if err := r.Client.Delete(ctx, mhc); err != nil && !apierrors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+
 		if controllerutil.ContainsFinalizer(nodePool, finalizer) {
 			controllerutil.RemoveFinalizer(nodePool, finalizer)
 			if err := r.Update(ctx, nodePool); err != nil {
 				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from nodePool: %w", err)
 			}
 		}
-		r.Log.Info("Deleted machineSet", "machineset", machineSet.GetName())
+		r.Log.Info("Deleted nodePool")
```

Comment:
nit: want to log the nodepool name?

Reply:
This is handled by controller runtime out of the box https://github.com/kubernetes-sigs/controller-runtime/blob/df2c43d8896d99e309f7da9d8fd16ad7a21f70a2/pkg/internal/controller/controller.go#L99-L101

Reply:
Ah! Nice, thanks